shell/app/modules/project/pages/release/components/form.tsx

```diff
@@ -177,84 +191,105 @@ const ReleaseForm = ({ readyOnly = false }: { readyOnly?: boolean }) => {
+                        {item.applicationName}
+                      </div>
+                    </div>
+                  </div>
+                  <span className="text-xs text-white-6">{moment(item.createdAt).format('YYYY/MM/DD HH:mm:ss')}</span>
                 </div>
-                <div className="text-xs flex">
-                  <div className="desc">{i18n.t('dop:owned application')}</div>
-                  <div className="ml-2 truncate" title={item.applicationName}>
-                    {item.applicationName}
+              );
+            },
+            rowKey: 'releaseId',
+            parentKey: 'applicationId',
+            menus: appList,
+            list: releaseList,
+            onMenuChange: selectApp,
+            onMenuFilter: searchApp,
+            onListFilter: searchRelease,
+            listPagination: {
+              total: releaseTotal,
+              current: pageNo,
+              pageSize: PAGINATION.pageSize,
+              onChange: (_pageNo: number) => {
+                setPageNo(_pageNo);
+                getReleases(_pageNo, appId);
+              },
+            },
+          },
+          readOnlyRender: (value: RELEASE.ReleaseDetail[]) => {
+            return (value || []).map((item: RELEASE.ReleaseDetail) => (
+              <div className="flex justify-between items-center bg-default-01 p-2">
+                <div>
+                  <div>{item.version}</div>
+                  <div className="text-xs flex mt-1">
+                    <div className="text-default-6">{i18n.t('dop:owned application')}</div>
+                    <div className="ml-2">{item.applicationName}</div>
                   </div>
                 </div>
+                <div className="text-default-6">{item.createdAt}</div>
               </div>
-              <span className="text-xs text-white-6">{moment(item.createdAt).format('YYYY/MM/DD HH:mm:ss')}</span>
-            </div>
-          );
-        },
-        rowKey: 'releaseId',
-        parentKey: 'applicationId',
-        menus: appList,
-        list: releaseList,
-        onMenuChange: selectApp,
-        onMenuFilter: searchApp,
-        onListFilter: searchRelease,
-        listPagination: {
-          total: releaseTotal,
-          current: pageNo,
-          pageSize: PAGINATION.pageSize,
-          onChange: (_pageNo: number) => {
-            setPageNo(_pageNo);
-            getReleases(_pageNo, appId);
+            ));
           },
+        }
+      : {
+          label: i18n.t('dop:upload files'),
+          name: 'diceFileID',
+          className: 'flex-nowrap',
+          getComp: (form: { form: FormInstance }) => <CustomUpload form={form} />,
+          rules: [
+            { required: true, message: i18n.t('dop:please upload files') },
+            {
+              validator: (_, value: string) => {
+                if (value === 'type-error') {
+                  return Promise.reject(new Error(i18n.t('dop:file type error')));
```

Comment:
is `new Error` required?

Reply:
The string effect is fine, but there will be an alarm.